listenbrainz/webserver/views/recommendations_cf_recording_api.py

```diff
@@ -0,0 +1,135 @@
+
+    artist_type = request.args.get('type')
+    if not _is_valid_artist(artist_type):
+        raise APIBadRequest("Invalid artist type: {}".format(artist_type))
+
+    recommendations = db_recommendations_cf_recording.get_user_recommendation(user['id'])
+
+    if recommendations is None:
+        err_msg = 'Oops! Looks like {} doesn\'t have recent listening history.' \
+                  ' Recommendations not generated :('.format(user_name)
+        raise APINotFound(err_msg)
+
+    if artist_type == 'top':
+        try:
+            payload = _get_top_artist_recommendations(recommendations, user_name)
+        except TopArtistRecommendationsNotFound:
+            err_msg = 'Oops! Looks like there was some Spark issue while generating top artist' \
+                      ' recommendations for {}. We are working on it. Please check back later!!'.format(user_name)
+            raise APINotFound(err_msg, payload={'last_updated': recommendations['created']})
+
+    if artist_type == 'similar':
+        try:
+            payload = _get_similar_artist_recommendations(recommendations, user_name)
+        except SimilarArtistRecommendationsNotFound:
+            err_msg = 'Oops! Looks like there was some Spark issue while generating similar artist' \
+                      ' recommendations for {}. We are working on it. Please check back later!!'.format(user_name)
+            raise APINotFound(err_msg, payload={'last_updated': recommendations['created']})
+
+    return jsonify(payload)
+
+
+def _get_top_artist_recommendations(recommendations, user_name):
+    """ Get top artist recommended recording mbids.
+
+        if recommendations not found raises TopArtistRecommendationsNotFound exception.
+    """
+    top_artist_recommendations = recommendations['recording_mbid']['top_artist']
+
+    if len(top_artist_recommendations) == 0:
+        raise TopArtistRecommendationsNotFound()
+
+    payload = {'payload': {
+        'top-artist': {
+            'recording_mbids': top_artist_recommendations
+        },
+        'user_name': user_name,
+        'last_updated': recommendations['created'],
+    }}
+
+    return payload
+
+
+def _get_similar_artist_recommendations(recommendations, user_name):
+    """ Get similar artist recommended recording mbids.
+
+        if recommendations not found raises SimilarArtistRecommendationsNotFound exception.
+    """
+    similar_artist_recommendations = recommendations['recording_mbid']['similar_artist']
+
+    if len(similar_artist_recommendations) == 0:
+        raise SimilarArtistRecommendationsNotFound()
+
+    payload = {'payload': {
+        'similar-artist': {
+            'recording_mbids': similar_artist_recommendations
+        },
+        'user_name': user_name,
+        'last_updated': recommendations['created'],
+    }}
+
+    return payload
+
+
+def _is_valid_artist(artist_type):
+    """ Check if artist type is valid.
+    """
+    if artist_type not in ['top', 'similar']:
```

Comment:
instead of keeping a list here, you should create an enum and return `artist_type in RecommendationArtistType`